src/screens/SignTransaction/signWithSD.ts

```diff
@@ -158,28 +159,43 @@ export const signTransactionWithSeedWords = async ({
   serializedPSBT,
   xfp,
   isMultisig,
+  isRemoteKey = false,
 }) => {
   try {
-    const inputs = idx(signingPayload, (_) => _[0].inputs);
+    const inputs = isRemoteKey
+      ? getInputsFromPSBT(serializedPSBT)
+      : idx(signingPayload, (_) => _[0].inputs);
+
     if (!inputs) throw new Error('Invalid signing payload, inputs missing');
-    const [signer] = defaultVault.signers.filter((signer) => signer.xfp === xfp);
+    const [signer] = isRemoteKey
+      ? [defaultVault.signers[0]]
+      : defaultVault.signers.filter((signer) => signer.xfp === xfp);
     const networkType = config.NETWORK_TYPE;
     // we need this to generate xpriv that's not stored
     const { xpub, xpriv } = generateSeedWordsKey(
       seedBasedSingerMnemonic,
       networkType,
       isMultisig ? EntityKind.VAULT : EntityKind.WALLET
     );
-    if (signer.xpub !== xpub) throw new Error('Invalid mnemonic; xpub mismatch');
+
+    const signerXpub = isRemoteKey
+      ? isMultisig
+        ? signer.signerXpubs[XpubTypes.P2WSH][0].xpub
+        : signer.signerXpubs[XpubTypes.P2WPKH][0].xpub
```

Comment:
Why is that though if the remote key is only for multisig?

Reply:
@ben-kaufman 
as mentioned earlier, Remote key is only for multisig.